cas/src/main/java/org/springframework/security/cas/web/authentication/DefaultServiceAuthenticationDetails.java

```diff
@@ -70,14 +71,13 @@ public String getServiceUrl() {
 
 	@Override
 	public boolean equals(Object obj) {
-		if (this == obj) {
```

Comment:
We want to preserve this because it a performance optimization for a frequently used piece of code.